spec/puppetserver/ca/config/puppet_spec.rb

```diff
@@ -338,4 +338,41 @@
       expect(conf.default_certname).to eq("foo")
     end
   end
+
+  it 'loads masterport correctly' do
+    masterport = 1234
+    Dir.mktmpdir do |tmpdir|
+      puppet_conf = File.join(tmpdir, 'puppet.conf')
+      File.open puppet_conf, 'w' do |f|
+        f.puts(<<-INI)
+          [server]
+            masterport = #{masterport}
+        INI
+      end
+
+      conf = Puppetserver::Ca::Config::Puppet.new(puppet_conf)
+      settings = conf.load(logger: logger)
+
+      expect(settings[:ca_port]).to eq("#{masterport}")
+      expect(stderr.string).to be_empty
+    end
+  end
+  it 'loads serverport correctly' do
```

Comment:
Would it be worth testing that `serverport` is preferred if both `masterport` and `serverport` are set? Or should we explicitly disallow that?

Reply:
While we do prefer serverport, I didn't test this because it wasn't something that might fail.